plugins/generator-1.21.1/neoforge-1.21.1/templates/armor.java.ftl

```diff
@@ -94,10 +94,11 @@ import net.minecraft.client.model.Model;
 		<#if data.bodyModelName != "Default" && data.getBodyModel()?? && data.enableBody>
 		event.registerItem(new IClientItemExtensions() {
 			@Override @OnlyIn(Dist.CLIENT) public HumanoidModel getHumanoidArmorModel(LivingEntity living, ItemStack stack, EquipmentSlot slot, HumanoidModel defaultModel) {
+				${data.bodyModelName} model = new ${data.bodyModelName}(Minecraft.getInstance().getEntityModels().bakeLayer(${data.bodyModelName}.LAYER_LOCATION));

```

Comment:
Would it be safe to even persist model as part of IClientItemExtensions instance to not create it every method call?

Reply:
possibly, I will test this right now to see

Reply:
It seems like this is possible, while initializing the model from inside getHumanoidArmorModel as the client extensions are constructed before model layers register

Reply:
Ok, so lazy loading. Sounds good! I have searched github and seems some mods do it like this, basically lazy-load the model/renderer if null

Reply:
Done! This should be a big improvement performance-wise over what was done before.